### DISCUSSION TITLE
Callback on existing exception_handler

### DISCUSSION THREAD

**Discussion by freddygarcia** [fastapi/fastapi#11071] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()


# For large projects, It would be useful to be able to handle what happens when an exception handler is duplicated.
app.on_exception_handler_duplicate = "error" # "ignore" 
app.on_exception_handler_duplicate = lambda exc: ...

# Module 1
app.add_exception_handler(ZeroDivisionError, lambda req, exc: "Zero division. Panic!")

# Module 2
# This is dangerous!!
app.add_exception_handler(ZeroDivisionError, lambda req, exc: "Chill out!")
```

### Description

On large projects divided by "components", "apps" or "modules" is important avoiding silently overriding  an exception handler.
The way I'm currently handling this is by extending the `add_exception_handler` and just logging when this scenario happens, but I'd be way better to have this feature on board.

My current approach is 

```

class FastAPI(BaseFastAPI):

    def add_exception_handler(
        self,
        exc_class_or_status_code: typing.Union[int, typing.Type[Exception]],
        handler: typing.Callable,
    ) -> None:
        
        if exc_class_or_status_code in self.exception_handlers:
            raise ValueError(
                f"An exception handler for {exc_class_or_status_code} was already registered"
            )
        
        self.exception_handlers[exc_class_or_status_code] = handler

```

### Operating System

Linux

### FastAPI Version

0.104.0

### Pydantic Version

2.5.3

### Python Version

3.11.7

**Comment by n8sty** [comment]
I don't understand the feature request. My suggestion would be what you're doing already, check if an exception handler is registered and then do _something_. I would say the opposite of this claim

> On large projects divided by "components", "apps" or "modules" is important avoiding silently overriding an exception handler.

is also true. It is also important, in large projects, to override exception handlers so that exceptions are handled uniformly at the level they should be (which I usually mean to be the outer-most part of the application.)

**Comment by YuriiMotov** [comment, marked as answer]
It's not so easy to implement this..
There is also a case that exception handler for status code will override exception handler for exception class:

```python
from fastapi import FastAPI, HTTPException
from fastapi.responses import JSONResponse

app = FastAPI()

class MyException(HTTPException):
    pass

@app.exception_handler(400)
async def status_handler(req, exc):
    return JSONResponse(content="Handled by status_handler", status_code=400)

@app.exception_handler(MyException)
async def my_exception_handler(req, exc):
    return JSONResponse(content="Handled by my_exception_handler", status_code=400)

@app.get("/")
async def root():
    raise MyException(detail="Error", status_code=400)

```
The code above will always return "Handled by status_handler", no matter which handler defined first.


Recommended approach is to cover this by tests.
